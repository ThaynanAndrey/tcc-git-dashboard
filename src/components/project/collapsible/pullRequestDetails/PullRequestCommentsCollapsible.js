import React from 'react';
import CollapsibleCard from '../../../layout/CollapsibleCard';

const styleCollapsibleHeader = {
    display: "flex",
    justifyContent: "space-between"
};

const styleCollapsibleHeaderText = {
    display: "flex"
};

/**
 * Stateless component with the function of presenting a Collapsible to Pull Requests.
 * 
 * @author Thaynan Nunes
 */
const PullRequestCommentsCollapsible = ({ pullRequest }) => {
    
    const getNumberStyles = () => ({
        width: "2rem",
        height: "27px",
        marginRight: "1rem",
        backgroundColor: (pullRequest.comments && pullRequest.comments.length > 0)
            ? "green" : "red",
        textAlign: "center",
        color: "white",
        borderRadius: "0.5rem"
    });

    const getItems = (items) => items && items.map((item, index) => (
        <tr key={index}>
            <td>{ item.author.name }</td>
            <td>{ item.date }</td>
            <td>{ item.message }</td>
        </tr>
    ));
    
    return (
        <CollapsibleCard isActive={false}>
            <div className="collapsible-header" style={styleCollapsibleHeader}>
                <div style={styleCollapsibleHeaderText}>
                    <div style={getNumberStyles()}>
                        {pullRequest.comments && pullRequest.comments.length}
                    </div>
                    Comentários
                </div>
            </div>
            { pullRequest.comments && pullRequest.comments.length > 0 &&
                <div className="collapsible-body body-collapsible-project">
                    <table className="striped highlight responsive-table">
                        <thead>
                            <tr>
                                <th>Responsável</th>
                                <th>Data Criação</th>
                                <th>Mensagem</th>
                            </tr>
                        </thead>
                        <tbody>
                            {getItems(pullRequest.comments)}
                        </tbody>
                    </table>
                </div>
            }
        </CollapsibleCard>
    );
};

export default PullRequestCommentsCollapsible;