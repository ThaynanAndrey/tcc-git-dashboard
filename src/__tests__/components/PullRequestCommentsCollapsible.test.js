import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import PullRequestCommentsCollapsible from '../../components/project/collapsible/pullRequestDetails/PullRequestCommentsCollapsible';

describe('Test PullRequestCommentsCollapsible', () => {
  let tree;
  
  beforeEach(() => {
    const pullRequest = {};

    tree = shallow(
      <PullRequestCommentsCollapsible
        pullRequest={pullRequest}
      />
    );
  });

  it('renders correctly a component to show PullRequestCommentsCollapsible', () => {
    expect(toJson(tree)).toMatchSnapshot();
  });

  
  it('the component should have class collapsible-header', () => {
    expect(tree.find(".collapsible-header").exists()).toBeTruthy();
  });

  it('the component should not have classes collapsible-body body-collapsible-project', () => {
    expect(tree.find(".collapsible-body.body-collapsible-project").exists()).toBeFalsy();
    expect(tree.find("table").exists()).toBeFalsy();
  });

  it('the component should have classes collapsible-body body-collapsible-project when pullRequest prop has comments', () => {
    const pullRequest = {
        comments: [{
            author: {name: "Teste"},
            date: "10/10/2010",
            message: "nada"
        }]
    };

    tree = shallow(
      <PullRequestCommentsCollapsible
        pullRequest={pullRequest}
      />
    );

    expect(tree.find(".collapsible-body.body-collapsible-project").exists()).toBeTruthy();
    expect(tree.find("table").exists()).toBeTruthy();
  });
})