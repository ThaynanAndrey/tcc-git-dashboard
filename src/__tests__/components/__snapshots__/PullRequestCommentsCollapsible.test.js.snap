// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PullRequestCommentsCollapsible renders correctly a component to show PullRequestCommentsCollapsible 1`] = `
<CollapsibleCard
  isActive={false}
>
  <div
    className="collapsible-header"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": "0.5rem",
            "color": "white",
            "height": "27px",
            "marginRight": "1rem",
            "textAlign": "center",
            "width": "2rem",
          }
        }
      />
      Coment√°rios
    </div>
  </div>
</CollapsibleCard>
`;
